examples/Dockerfile

```diff
@@ -1,18 +1,29 @@
-FROM jboss/base-jdk:7
+FROM jboss/base-jdk:8
 
 ENV KEYCLOAK_VERSION 1.9.0.Final
 ENV JBOSS_HOME /opt/jboss/keycloak-demo/keycloak
+ENV MAVEN_VERSION 3.3.9
 
 USER root
 
-RUN yum -y install maven && yum clean all
+RUN yum -y install wget && yum clean all
+
+RUN curl -fsSL https://archive.apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz | tar xzf - -C /usr/share \
+  && mv /usr/share/apache-maven-$MAVEN_VERSION /usr/share/maven \
+  && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
+
+ENV MAVEN_HOME /usr/share/maven
 
 USER jboss
 
-RUN cd /opt/jboss && curl -L http://sourceforge.net/projects/keycloak/files/$KEYCLOAK_VERSION/keycloak-demo-$KEYCLOAK_VERSION.tar.gz/download | tar zx && mv /opt/jboss/keycloak-demo-$KEYCLOAK_VERSION /opt/jboss/keycloak-demo
+RUN cd /opt/jboss && curl -s http://downloads.jboss.org/keycloak/$KEYCLOAK_VERSION/keycloak-demo-$KEYCLOAK_VERSION.zip -o tmp.zip && unzip tmp.zip -d . && mv /opt/jboss/keycloak-demo-$KEYCLOAK_VERSION /opt/jboss/keycloak-demo
 
 RUN mvn package -f /opt/jboss/keycloak-demo/examples/preconfigured-demo/pom.xml && rm -rf ~/.m2/repository
 RUN cd /opt/jboss/keycloak-demo/examples/preconfigured-demo && find -name *.war | grep -v ear | xargs -I {} cp {} /opt/jboss/keycloak-demo/keycloak/standalone/deployments/ && cp /opt/jboss/keycloak-demo/examples/preconfigured-demo/testrealm.json /opt/jboss/keycloak-demo/keycloak/
 
-EXPOSE 8080
-CMD ["/opt/jboss/keycloak-demo/keycloak/bin/standalone.sh", "-b", "0.0.0.0", "-Dkeycloak.import=/opt/jboss/keycloak-demo/keycloak/testrealm.json"]
+EXPOSE 8080 9990
```

Comment:
None of the management consoles should be exposed by default. You cannot "unexpose" a port in Docker. Since this is a general use image - only strictly required ports should be exposed. If someone wants to connect to console - it should be exposed by hand.